src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php

```diff
@@ -246,6 +237,79 @@ public static function register($mode = 0)
         }
     }
 
+    private static function isLaggingVendor(array $trace): bool
+    {
+        foreach ($trace as $line) {
+            if (isset($line['file'])) {
+                $file = $line['file'];
+                if ('-' === $file) {
+                    continue;
+                }
+                if (!self::inVendors($file)) {
+                    return false;
+                }
+                if (isset($erroringFile, $erroringPackage)) {
+                    if (self::getPackage($file) !== $erroringPackage) {
+                        return true;
+                    }
+                } else {
+                    $erroringFile = $file;
+                    $erroringPackage = self::getPackage($file);
+                }
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * inVendors() should always be called prior to calling this method.
+     */
+    private static function getPackage(string $path): string
+    {
+        $path = realpath($path) ?: $path;
+        foreach (self::getVendors() as $vendorRoot) {
+            if (0 === strpos($path, $vendorRoot)) {
+                $relativePath = substr($path, strlen($vendorRoot) + 1);
+                $vendor = strstr($relativePath, DIRECTORY_SEPARATOR, true);
+
+                return $vendor.'/'.strstr(substr($relativePath, strlen($vendor) + 1), DIRECTORY_SEPARATOR, true);
+            }
+        }
```

Comment:
you should throw a meaningful exception when reaching this place explaining that the provided path is not a vendor path, instead of letting PHP throw a fatal error due to `null` not respecting the `string` return type (which would not be clear at all).

Reply:
@stof we're inside the error handler, that's why I refrained from doing that, I'm not sure how it will behave, but I will try.

Reply:
Wait, it's an *error* handler, so an exception should probably be fine.

Reply:
Reaching this place would indicate a bad usage of the private method inside the class anyway (as you are not supposed to pass a non-vendor path). So I think it would be fine (and would make it easier for contributors to see their mistakes)

Reply:
I tried putting an exception unconditionally on the first line, and it worked as expected, so it's fine indeed :+1: